dig.go

```diff
@@ -108,6 +117,20 @@ func Name(name string) ProvideOption {
 	})
 }
 
+// Group is a ProvideOption that specifies that all values produced by a
+// constructor should be members of the same group. Since all members of a
+// group must be of same type, this option cannot be provided for constructors
```

Comment:
This is incorrect. You can totally do,

```
type Result struct {
  fx.Out

  Reader io.Reader `group:"foo"`
  Writer io.Writer `group:"foo"`
}
```

Consumption,

```
type Param struct {
  fx.In

  Readers []io.Reader `group:"foo"`
  Writers []io.Writer `group:"foo"`
}
```

The primary-key is the triple `(type, value name, group name)` with the
additional constraint that only one of "value name" and "group name" is set.